src/js-form-validator.test.ts

```diff
@@ -85,3 +85,31 @@ QUnit.test('should fail email validation with special characters in domain', ass
 
   assert.equal(FormValidator.Validate(inputElement), false);
 });
+
+QUnit.test('should fail checkbox validation when checked state is not set', assert => {
+  inputElement.setAttribute('type', 'checkbox');
+  inputElement.setAttribute('data-validate', 'checkbox');
+
+  assert.equal(FormValidator.Validate(inputElement), false);
+});
+
+QUnit.test('should fail checkbox validation when checked state set to false', assert => {
```

Comment:
It looks like this will fail validation if the checkbox is not checked. Is the context for this an "I have read and agree to the terms of service" type of checkbox? Is there some way to clarify that in the test so people will know that in other situations an empty checkbox will pass validation.

Reply:
@samiam236 good catch. I added a check to the `data-empty_allowed` attribute, and added a couple tests.